plugins/query-insights/src/main/java/org/opensearch/plugin/insights/core/listener/QueryInsightsListener.java

```diff
@@ -138,10 +147,24 @@ public void onRequestEnd(final SearchPhaseContext context, final SearchRequestCo
             attributes.put(Attribute.TOTAL_SHARDS, context.getNumShards());
             attributes.put(Attribute.INDICES, request.indices());
             attributes.put(Attribute.PHASE_LATENCY_MAP, searchRequestContext.phaseTookMap());
+
+            // Get internal computed and user provided labels
+            Map<String, Object> labels = new HashMap<>();
+            // Retrieve user provided label if exists
+            String userProvidedLabel = getUserProvidedTag(threadPool);
+            if (userProvidedLabel != null) {
+                labels.put(Task.X_OPAQUE_ID, userProvidedLabel);
+            }
+            attributes.put(Attribute.LABELS, labels);
```

Comment:
I was just checking to see how slow logs read the `X-Opaque-id` header, and it looks like it doesn't need to access the `ThreadContext` directly, since the headers are associated with the current `Task`: https://github.com/opensearch-project/OpenSearch/blob/a5f335f94b135bf9e7f921f7b3859a019d910f22/server/src/main/java/org/opensearch/action/search/SearchRequestSlowLog.java#L203-L221

We shouldn't need to pass `ThreadPool` to the `QueryInsightsListener`, since we can just read the headers from the task. (Not sure if this works for the resource tracking headers too. Maybe?)